### PR TITLE
e2e_common: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

e2e_common: stop using deprecated framework.ExpectEqual

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/common/node/container_probe.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ var _ = SIGDescribe("Probing container", func() {
 		}
 
 		restartCount := getRestartCount(p)
-		framework.ExpectEqual(restartCount, 0, "pod should have a restart count of 0 but got %v", restartCount)
+		gomega.Expect(restartCount).To(gomega.BeIdenticalTo(0), "pod should have a restart count of 0 but got %v", restartCount)
```
Comment (endocrimes) [inline test/e2e/common/node/container_probe.go:98]: I'm curious as to the reasoning behind the `BeIdenticalTo` usage here - Non-standard `Equal` checks kinda draw attention to themselves imo?

Comment (carlory) [reply test/e2e/common/node/container_probe.go:98]: @endocrimes 

In this case, their functions are the same. The only difference is that BeIdenticalTo uses the `==` to compare, while Equal uses `reflect.DeepEqual` to compare. Relatively speaking, the performance of BeIdenticalTo may be better. Of course, if this is not considered, Equal is also reasonable for a unified style.

Comment (endocrimes) [reply test/e2e/common/node/container_probe.go]: I understand the behavior difference :) - My question was more about the reason behind using them in this particular case, when not trying to test for pointer equality explicitly 😅   - Definitely would rather stick to a unified style and stick to using exceptions only when they're needed

File: test/e2e/common/node/pods.go [hunk 2]
```diff
@@ -997,8 +997,8 @@ var _ = SIGDescribe("Pods", func() {
 		ginkgo.By("getting the Pod and ensuring that it's patched")
 		pod, err := f.ClientSet.CoreV1().Pods(testNamespaceName).Get(ctx, testPodName, metav1.GetOptions{})
 		framework.ExpectNoError(err, "failed to fetch Pod %s in namespace %s", testPodName, testNamespaceName)
-		framework.ExpectEqual(pod.ObjectMeta.Labels["test-pod"], "patched", "failed to patch Pod - missing label")
-		framework.ExpectEqual(pod.Spec.Containers[0].Image, testPodImage2, "failed to patch Pod - wrong image")
+		gomega.Expect(pod.ObjectMeta.Labels["test-pod"]).To(gomega.Equal("patched"), "failed to patch Pod - missing label")
```
Comment (pohly) [inline test/e2e/common/node/pods.go:1000]: Can be enhanced with `HaveKeyWithValue`.

File: test/e2e/common/node/podtemplates.go [hunk 3]
```diff
@@ -99,7 +100,7 @@ var _ = SIGDescribe("PodTemplates", func() {
 		// get template (ensure label is there)
 		podTemplateRead, err = f.ClientSet.CoreV1().PodTemplates(testNamespaceName).Get(ctx, podTemplateName, metav1.GetOptions{})
 		framework.ExpectNoError(err, "failed to get PodTemplate")
-		framework.ExpectEqual(podTemplateRead.ObjectMeta.Labels["podtemplate"], "patched", "failed to patch template, new label not found")
+		gomega.Expect(podTemplateRead.ObjectMeta.Labels["podtemplate"]).To(gomega.Equal("patched"), "failed to patch template, new label not found")
```
Comment (pohly) [inline test/e2e/common/node/podtemplates.go:103]: Can be enhanced with `HaveKeyWithValue`.

File: test/e2e/common/node/runtimeclass.go [hunk 4]
```diff
@@ -281,24 +282,24 @@ var _ = SIGDescribe("RuntimeClass", func() {
...
 		rcs, err := rcClient.List(ctx, metav1.ListOptions{LabelSelector: "test=" + f.UniqueName})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(len(rcs.Items), 3, "filtered list should have 3 items")
+		gomega.Expect(rcs.Items).To(gomega.HaveLen(3), "filtered list should have 3 items")
 
 		ginkgo.By("patching")
 		patchedRC, err := rcClient.Patch(ctx, createdRC.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(patchedRC.Annotations["patched"], "true", "patched object should have the applied annotation")
+		gomega.Expect(patchedRC.Annotations["patched"]).To(gomega.Equal("true"), "patched object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/common/node/runtimeclass.go:295]: Can be enhanced with `HaveKeyWithValue`.

**Review by pohly — CHANGES_REQUESTED** [review]
Looks good overall, some suggestions.

**Comment by carlory** [comment]
@pohly please review again.

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**